api/v2/performanceprofile_validation_test.go

```diff
@@ -292,11 +284,34 @@ var _ = Describe("PerformanceProfile", func() {
 	})
 
 	Describe("Net validation", func() {
-		It("should have net fields properly populated", func() {
-			errors := profile.validateNet()
-			Expect(errors).To(BeEmpty(), "should not have validation errors with properly populated net devices fields")
+		Context("with properly populated fields", func() {
+			It("should have net fields properly populated", func() {
+				errors := profile.validateNet()
+				Expect(errors).To(BeEmpty(), "should not have validation errors with properly populated net devices fields")
+			})
+		})
+		Context("with misconfigured fields", func() {
+			It("should raise the validation syntax errors", func() {
+				invalidVendor := "123"
+				invalidModel := "0x12345"
+				profile.Spec.Net.Devices[0].InterfaceName = pointer.StringPtr("")
+				profile.Spec.Net.Devices[0].VendorID = pointer.StringPtr(invalidVendor)
+				profile.Spec.Net.Devices[0].ModelID = pointer.StringPtr(invalidModel)
+				errors := profile.validateNet()
+				Expect(errors).NotTo(BeEmpty())
+				Expect(errors[0].Error()).To(ContainSubstring(fmt.Sprintf("device name cannot be empty")))
+				Expect(errors[0].Error()).To(ContainSubstring(fmt.Sprintf("device vendor ID %s has an invalid format. Vendor ID should be represented as 0x<4 hexadecimal digits> (16 bit representation)", invalidVendor)))
+				Expect(errors[0].Error()).To(ContainSubstring(fmt.Sprintf("device model ID %s has an invalid format. Model ID should be represented as 0x<4 hexadecimal digits> (16 bit representation)", invalidModel)))
```

Comment:
should not be checking *at least one* between InterfaceName vs VendorID+ModelID should be given? (this should be an additional test probably)